### PR TITLE
[NUI] Add ExcludeLayouting to View

### PR DESCRIPTION
**Pull request by bshsqa**
- If ExcludeLayouting property is false, the view is not affected by parent layout.
 - Even the property is false, the children Views are affected by layout of current View.
 - This ExcludeLayouting property is changed synchronously with SetPositionByLayout of its Layout.
 - Children of AbsoluteLayout can freely be animated.

### CODE REVIEW

**Comment by dasg34** [comment]
`WidthSpecification`(`HeightSpecification`) does not work when `SetPositionByLayout` is `false`.
Please check example below.
```cs
            View root = new View();
            root.BackgroundColor = new Color(255 / 255f, 255 / 255f, 227 / 255f, 1.0f);
            root.Layout = new AbsoluteLayout();
            root.WidthSpecification = 600;
            root.HeightSpecification = 600;
            Window.Instance.GetDefaultLayer().Add(root);

            Button button = new Button();
            button.WidthSpecification = 100;
            button.HeightSpecification = 100;
            button.Text = "btn";
            root.Add(button);

            View absoluteLayout = new View();
            absoluteLayout.Layout = new AbsoluteLayout();
            absoluteLayout.Layout.SetPositionByLayout = false;
            absoluteLayout.WidthSpecification = LayoutParamPolicies.WrapContent;
            absoluteLayout.HeightSpecification = 300;
            absoluteLayout.Position2D = new Position2D(100, 100);
            absoluteLayout.BackgroundColor = Color.Yellow;
            root.Add(absoluteLayout);

            Button absoluteLayoutButton = new Button();
            absoluteLayoutButton.WidthSpecification = 100;
            absoluteLayoutButton.HeightSpecification = 100;
            absoluteLayoutButton.BackgroundColor = Color.Blue;
            absoluteLayoutButton.Text = "Button";
            absoluteLayout.Add(absoluteLayoutButton);
```
| DevelNUI branch | pr/2187 branch |
| --- | ---|
|before |after |

**Comment by dasg34** [comment]
hmm,, i'm not sure when to use this property.
If you don't want the view to be affected by parent layout, you can add the view directly to `Window.Instance.GetDefaultLayer()`.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 1]
```diff
@@ -205,6 +206,23 @@ protected set
...
+            get
+            {
+                return transformedByLayout;
+            }
+            set
+            {
+                transformedByLayout = value;
+                if(Layout != null && Layout.SetPositionByLayout != value)
+                {
+                    Layout.SetPositionByLayout = value;
```
Comment (dasg34) [inline src/Tizen.NUI/src/public/BaseComponents/View.cs:221]: I think `Layout.RequestLayout()` needs to be called for re-layout when `SetPositionByLayout` value is changed.

File: src/Tizen.NUI.Components/Controls/ScrollableBase.cs [hunk 2]
```diff
@@ -433,10 +433,6 @@ public new LayoutItem Layout
             set
             {
                 ContentContainer.Layout = value;
-                if (ContentContainer.Layout != null)
```
Comment (dasg34) [inline src/Tizen.NUI.Components/Controls/ScrollableBase.cs:436]: Is this change related to `TransformedByLayout` property? 
This seems to change the behavior of `ScrollableBase`.

File: src/Tizen.NUI/src/public/Layouting/FlexLayout.cs [hunk 3]
```diff
@@ -702,25 +710,40 @@ protected override void OnMeasure(MeasureSpecification widthMeasureSpec, Measure
...
             int count = LayoutChildren.Count;
             for (int childIndex = 0; childIndex < count; childIndex++)
             {
                 LayoutItem childLayout = LayoutChildren[childIndex];
                 if (childLayout != null)
                 {
-                    // Get the frame for the child, start, top, end, bottom.
-                    Vector4 frame = new Vector4(Interop.FlexLayout.FlexLayout_GetNodeFrame(swigCPtr, childIndex), true);
-                    childLayout.Layout(new LayoutLength(frame.X), new LayoutLength(frame.Y), new LayoutLength(frame.Z), new LayoutLength(frame.W));
+                    if (!childLayout.Owner.TransformedByLayout)
```
Comment (dasg34) [inline src/Tizen.NUI/src/public/Layouting/FlexLayout.cs:729]: What do you think of using `SetFlexPositionType`?(https://yogalayout.com/docs/absolute-relative-layout)
Since the child with TransformedByLayout(false) is already added in children of yoga-layout at `InsertChild()`, i'm not sure that FlexLayout_CalculateLayout does work properly.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs (hunk 1 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/BaseComponents/View.cs:221]: Thank you it is needed.

File: src/Tizen.NUI.Components/Controls/ScrollableBase.cs (hunk 2 above)
Comment (bshsqa) [reply src/Tizen.NUI.Components/Controls/ScrollableBase.cs:436]: You are right. I fixed it with the first comment above.

File: src/Tizen.NUI/src/public/Layouting/FlexLayout.cs (hunk 3 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Layouting/FlexLayout.cs]: Thanks, I used it and tested. Well works.

File: src/Tizen.NUI/src/public/Layouting/AbsoluteLayout.cs [hunk 4]
```diff
@@ -54,6 +54,11 @@ protected override void OnMeasure(MeasureSpecification widthMeasureSpec, Measure
                     // Get size of child with no padding, no margin. we won't support margin, padding for AbsolutLayout.
                     MeasureChildWithoutPadding(childLayout, widthMeasureSpec, heightMeasureSpec);
 
+                    if(!childLayout.Owner.TransformedByLayout)
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Layouting/AbsoluteLayout.cs:57]: I found there are a lot of `if(...)`(no space) in you patch here and I think it is because it is already used in `LinearLayout.cs`.
But I think it would be better to use `if (...)` at least to the newly added code.

[the remark this answers is not in the document]
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Layouting/AbsoluteLayout.cs]: Thank you I changed it for all (except View.cs) patch codes

File: src/Tizen.NUI/src/public/Layouting/LinearLayout.cs [hunk 5]
```diff
@@ -726,27 +747,30 @@ private void LayoutHorizontal(LayoutLength left, LayoutLength top, LayoutLength
...
-                            break;
-                        case Alignment.Top: // FALLTHROUGH default
-                        default:
-                            childTop = new LayoutLength(Padding.Top + childMargin.Top);
-                            break;
+                        LayoutLength childWidth = childLayout.MeasuredWidth.Size;
+                        LayoutLength childHeight = childLayout.MeasuredHeight.Size;
+                        Extents childMargin = childLayout.Margin;
+
+                        switch ( LinearAlignment )
```
[the remark this answers is not in the document]
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Layouting/LinearLayout.cs]: Thank you I changed it for all (except View.cs) patch codes

**Review by dasg34 — APPROVED** [review]
LGTM.

**Review by Seoyeon2Kim — APPROVED** [review]
Looks good to me, but I think it would be better to get approvals from others as well.

**Review by AdunFang — APPROVED** [review]
It is OK.
